### ISSUE TITLE
a11y(iOS): nested `<Text role="link" />` not focusable on iOS

### ISSUE THREAD

**Issue by mdjastrzebski** [facebook/react-native#53243]
### Description

`<Text role="link" />` element nested in `<Text />` is not focusable by screen reader on iOS.

### Example code
```tsx
<Text style={styles.text}>
  Regular Text
  <Text role="link" onPress={handlePress}>Link Text</Text>
</Text>
```

On iOS it focuses only the top-level `Text` element, there is no focus on child text element with link role.

on Android it focuses first on top-level `Text` element (announcing: "contains links"), then focuses on child text element with link role.

This is especially hard to solve, if I would want to have 2+ links inside the text.

### Steps to reproduce

1. Clone https://github.com/mdjastrzebski/ReproLinkText
2. Run under screen reader on iOS (& Android for comparison)

### React Native Version

0.81.0

### Affected Platforms

Runtime - iOS

### Output of `npx @react-native-community/cli info`

```text
info Fetching system and libraries information...
System:
  OS: macOS 15.6
  CPU: (11) arm64 Apple M3 Pro
  Memory: 183.14 MB / 36.00 GB
  Shell:
    version: "5.9"
    path: /bin/zsh
Binaries:
  Node:
    version: 22.17.1
    path: ~/.nvm/versions/node/v22.17.1/bin/node
  Yarn:
    version: 3.6.4
    path: ~/.nvm/versions/node/v22.17.1/bin/yarn
  npm:
    version: 10.9.2
    path: ~/.nvm/versions/node/v22.17.1/bin/npm
  Watchman:
    version: 2025.07.28.00
    path: /opt/homebrew/bin/watchman
Managers:
  CocoaPods:
    version: 1.15.2
    path: /Users/mdj/.rbenv/shims/pod
SDKs:
  iOS SDK:
    Platforms:
      - DriverKit 24.5
      - iOS 18.5
      - macOS 15.5
      - tvOS 18.5
      - visionOS 2.5
      - watchOS 11.5
  Android SDK:
    API Levels:
      - "35"
      - "36"
    Build Tools:
      - 35.0.0
      - 36.0.0
    System Images:
      - android-35 | Google Play ARM 64 v8a
      - android-36 | Google Play ARM 64 v8a
    Android NDK: Not Found
IDEs:
  Android Studio: 2025.1 AI-251.25410.109.2511.13752376
  Xcode:
    version: 16.4/16F6
    path: /usr/bin/xcodebuild
Languages:
  Java:
    version: 17.0.16
    path: /usr/bin/javac
  Ruby:
    version: 3.4.5
    path: /Users/mdj/.rbenv/shims/ruby
npmPackages:
  "@react-native-community/cli":
    installed: 20.0.0
    wanted: 20.0.0
  react:
    installed: 19.1.0
    wanted: 19.1.0
  react-native:
    installed: 0.81.0
    wanted: 0.81.0
  react-native-macos: Not Found
npmGlobalPackages:
  "*react-native*": Not Found
Android:
  hermesEnabled: true
  newArchEnabled: true
iOS:
  hermesEnabled: true
  newArchEnabled: false
```

### Stacktrace or Logs

```text
N/A
```

### MANDATORY Reproducer

https://github.com/mdjastrzebski/ReproLinkText

### Screenshots and Videos

iOS (incorrect):
https://github.com/user-attachments/assets/623b4a6d-2ba2-4a7d-a659-7f16386c7ca2

Android (correct):
https://github.com/user-attachments/assets/216921f8-6f59-4261-b75f-574a7cb1de0e

**Comment by mdjastrzebski** [comment]
not stale

**Comment by tomvanhees** [comment]
The same on android as well.
